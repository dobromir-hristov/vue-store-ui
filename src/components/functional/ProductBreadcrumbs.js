import ProductBreadcrumbsItem from '@/components/design/ProductBreadcrumbsItem'

export const ProductBreadcrumbs = {
  components: { ProductBreadcrumbsItem },
  props: {
    items: {
      type: Array,
      default: () => ([])
    }
  }
}
