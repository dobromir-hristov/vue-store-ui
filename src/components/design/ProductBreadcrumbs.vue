<template>
  <div class="ProductBreadcrumbs flex mb-4">
    <div
      v-for="(item, index) in items"
      :key="item.id"
      class="ProductBreadcrumb mr-2 flex text-grey"
    >
      <product-breadcrumbs-item
        :is-first="index === 0"
        :item="item"
      />
    </div>
  </div>
</template>

<script>
import { ProductBreadcrumbs } from '@/components/functional/ProductBreadcrumbs'

/**
 * Minimalistic breadcrumbs navigation
 * @module ProductBreadcrumbs
 */
export default {
  name: 'ProductBreadcrumbs',
  mixins: [ProductBreadcrumbs]
}
</script>
