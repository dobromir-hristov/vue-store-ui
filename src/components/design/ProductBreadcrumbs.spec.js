import ProductBreadcrumbs from './ProductBreadcrumbs'

/**
 * @param overrides
 * @returns {{config, wrapper: Wrapper<V> | Wrapper<Vue> | *}}
 */
const createWrapper = (overrides) => {
  let config = _merge(createComponentMocks({ router: true }),
    {
      propsData: {
        items: [
          { id: '2', name: 'Default Category' },
          { id: '3', name: 'Women' }
        ]
      }
    },
    overrides
  )
  const wrapper = mount(ProductBreadcrumbs, config)
  return {
    config,
    wrapper
  }
}

describe('ProductBreadcrumbs', () => {
  it('renders the ProductBreadcrumbs', () => {
    const { wrapper } = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
