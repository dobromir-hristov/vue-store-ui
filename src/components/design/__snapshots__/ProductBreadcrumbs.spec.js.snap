// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBreadcrumbs renders the ProductBreadcrumbs 1`] = `
<div class="ProductBreadcrumbs flex mb-4">
  <div class="ProductBreadcrumb mr-2 flex text-grey">
    <div class="ProductBreadcrumbsItem">
      <router-link-stub to="[object Object]" class="mr-2 text-grey-dark ml-0">
        Default Category
      </router-link-stub>
      /
    </div>
  </div>
  <div class="ProductBreadcrumb mr-2 flex text-grey">
    <div class="ProductBreadcrumbsItem">
      <router-link-stub to="[object Object]" class="mr-2 text-grey-dark">
        Women
      </router-link-stub>
      /
    </div>
  </div>
</div>
`;
